### ISSUE TITLE
[Bug][backend core] more readable error message should be display in all HTTP APIs

### ISSUE THREAD

**Issue by likyh** [apache/incubator-devlake#3467]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

![image](https://user-images.githubusercontent.com/3294100/196390301-bca102e5-d603-45f1-9978-24e05067457a.png)
![image](https://user-images.githubusercontent.com/3294100/196390373-dd81bda2-d671-4bbf-a7ff-15c69288f30c.png)

Wraps: (4) Your token has not been granted the required scopes to execute this query. The 'email' field requires one of the following scopes: ['user:email', 'read:user'], but your token has only been granted the: ['repo'] scopes. Please modify your token's scopes at: https://github.com/settings/tokens.

The most important log not show in UI.

### What you expected to happen

more readable error message should be display in all HTTP APIs

### How to reproduce

1. create a blueprint with github_graphql
2. run someway to got an error
3. look at config-UI

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by keon94** [comment]
Could you check the payload response to see if the root cause message is contained in the "causes" array?

**Comment by likyh** [comment]
> What's the full string in the "message" field?

Some are right:
```
subtask extractApiPrReviewComments ended unexpectedly
caused by: error calling plugin Extract implementation (404)
caused by: parse prId failed (404)
caused by: github pull request not found in DB (404)
caused by: record not found
```

Some are wrong:
```
run task failed with panic (github.com/apache/incubator-devlake/plugins/github_graphql/tasks.CollectRepo.func2:96)
```
![image](https://user-images.githubusercontent.com/3294100/196489631-fdaf2faf-a475-4e7b-a5bb-1046be73c15e.png)

**Comment by keon94** [comment]
That looks like a different error. I was expecting the payload response from the failed GraphQL call. That has to contain the Github error per my expectation.

**Comment by klesh** [comment]
Sounds like we ought to store the `causes` in the database as a json array?

**Comment by Startrekzky** [comment]
@likyh  Is this issue fixed by #3601 ?

**Comment by likyh** [comment]
> @likyh Is this issue fixed by #3601 ?

A part of them. And others may be fixed in the next issue found.
